### ISSUE TITLE
Client Error: bad object in message for a very specific find query

### ISSUE THREAD

**Issue by xprt64** [mongodb/mongo-php-library#449]
There is a special find query that fails with the message: `Client Error: bad object in message: bson length doesn't match what we found in object with unknown _id`

One can change the characters but not the length of strings and obtain the same exception.
If the order of fields is swapped then the exception is not thrown anymore.

The UnitTest I used to reproduce is this:

```
use MongoDB\BSON\ObjectID;
use MongoDB\BSON\Regex;

class SpecialFindTest extends FunctionalTestCase
{
    /** should not throw exception */
    public function testSpecification()
    {
        $this->collection->findOne([
            '_______'                             => new ObjectID('111111111111111111111111'),
            '___________________________________' => new Regex('_______________________________________________________', 'i'),
        ]);
    }
}
```

**Environment:**
MongoDB extension version: 1.3.4
libbson bundled version: 1.8.2
libmongoc bundled version: 1.8.2
mongodb/mongodb: 1.2.0
PHP 7.1.11-0ubuntu0.17.10.1 (cli) (built: Nov  1 2017 16:30:52) ( NTS )
PHP 7.1.3 (cli) (built: Mar 21 2017 23:12:30) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.1.0, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.1.3, Copyright (c) 1999-2017, by Zend Technologies

**Comment by jmikola** [comment]
I've opened [PHPC-1067](https://jira.mongodb.org/browse/PHPC-1067) to track this, and added some other observations to that ticket. Feel free to follow it for progress.

**Comment by derickr** [comment]
This ended up being a bug in libbson: https://jira.mongodb.org/browse/CDRIVER-2455

**Comment by xprt64** [comment]
In my production environment, in 400,000 of inserts only 1 produced this error.

**Comment by jmikola** [comment]
[PHPC-1067](https://jira.mongodb.org/browse/PHPC-1067) (our tracking ticket for [CDRIVER-2455](https://jira.mongodb.org/browse/CDRIVER-2455)) was released in version [1.4.0](https://pecl.php.net/package/mongodb/1.4.0) of the extension. This can be closed.
